src/map/atcommand.cpp

```diff
@@ -701,6 +701,69 @@ ATCE atcommand_goto(Session *s, dumb_ptr<map_session_data> sd,
 }
 
 static
+ATCE atcommand_npc(Session *s, dumb_ptr<map_session_data> sd,
+        ZString message)
+{
+    NpcName npc;
+
+    if (!asplit(message, &npc))
+    {
+        clif_displaymessage(s,
+                "Please, enter a npc name (usage: @npc/@warptonpc/@gotonpc <npc>)."_s);
+        return ATCE::USAGE;
+    }
+
+    dumb_ptr<npc_data> nd = npc_name2id(npc);
+    if (nd != nullptr)
+    {
+        if (nd->bl_m->flag.get(MapFlag::NOWARPTO)
+            && !(pc_isGM(sd).satisfies(battle_config.any_warp_GM_min_level)))
+        {
+            clif_displaymessage(s,
+                    "You are not authorised to warp you to the map of this npc."_s);
+            return ATCE::PERM;
+        }
+        if (sd->bl_m->flag.get(MapFlag::NOWARP)
+            && !(pc_isGM(sd).satisfies(battle_config.any_warp_GM_min_level)))
+        {
+            clif_displaymessage(s,
+                    "You are not authorised to warp you from your actual map."_s);
+            return ATCE::PERM;
+        }
+
+        int x = nd->bl_x, y = nd->bl_y, x0 = (x >= 3)? (x - 3): 0, j = 0,
```

Comment:
Well we don't put collision on most our NPCs but I know some, like Hetchel, have collision. anything else I don't think is supposed to be reachable. such as mob spawns or XmasSpawnManager.